### DISCUSSION TITLE
SEBE: time-offset

### DISCUSSION THREAD

**Discussion by MatLudvik** [UMEP-dev/UMEP#333] [Q&A] [answered]
Hello,

I would like to ask about one uncertainty in time-offset while using SEBE. I am currently trying to obtain the radiation using data from Austria [UTC+1/UTC+2 (due to half a year time-shift in CET)]. In the first result I got, the radiation was coming suspiciously too much from the east side. The buildings on the southwest side were getting much less radiation. Personally, I would attribute it to an error in the use of time-offset.
I used:
- meteorological data ERA5 from Copernicus obtained from pre-processor in UMEP (1 year period)
- offset UTC +2 (then also UTC+1)
- diffuse and direct shortwave ticked to have the right result

Do you use time-offset even while using meteo-data from ERA5 based on site coordinates? (While using no time-offset, the result looked fine, with radiation coming from the south)
How would you solve the uncertainty in the time-shift in CET time?
Where do you think I might possibly have gone wrong?

Thank you in advance for your insights/help/discussion.

**Comment by biglimp** [comment]
SEBE (and all other models in UMEP) make use of local time. 
Maybe ERA5 data is in UTC? I dont remember. How do you accuire the ERA5 data? I recommend that you plot the solar radiation data from ERA5 and investigate when solar maximum appears for each day.

**Comment by MatLudvik** [comment]
I accouired the ERA5 data directly from UMEP using Download ERA5 Data tool. The data are in local time (UTC+1).
Is this feature (time-offset) working just with the meteorological data as an input?

So do I get it right, that if the ERA5 data are already in local time, there is no need to fill in any time-offset, because it would shift the meteorological-data coming from more east? Would you fill in the time-offset from UTC just in case the meteorological data are in UTC?

Thank you

**Comment by biglimp** [comment]
Ok.
The ERA5 data downloaded using UMEP will be fine for SEBE, as long as it is in local time.
You need to fill in the correct utc for your location as well have the data in local time.

**Comment by MatLudvik** [comment]
Thanks for yours quick response.
I was probably wrong. If the shortwave radiation values are indeed average over previous 1 hour, the data downloaded from Copernicus are in UTC.

I looked at your demo data (should be in LST I suppose?) and compared to the meteorological data that can be retrieved from Copernicus using Download ERA5 Data tool in UMEP there is indeed a shift. So if the time-offset feature in SEBE tool makes just a time shift in meteorological data and I get the data from Download ERA5 Data tool in UMEP in UTC, I need either (if I get it right):
 1) not fill in the time-offset in SEBE
 2) change the time in the whole meteorological file to the local time and then I can fill in the offset from UTC.

I'm still struggling with the fact that if I don't do as I say and stick to the tutorial using local time UTC+1 (while using data from Copernicus), I have the largest amount of incoming radiation coming strongly from the southeast, which doesn't correspond the assumed reality.

Please let me know if I am wrong, but I have no other explanation for that

**Comment by biglimp** [comment, marked as answer]
What demo dat are you referring to, from Gothenburg? Then it is local time. I still recommend that you plot the met data and adjust your utc based on that. Maybe you have found a bug. Submit your data (dsm and metdata) and I can have a look.

**Reply by MatLudvik** [reply]
Hi, 
yes, I referred to the data from Gothenburg. The meteo data that you can download from Copernicus using Download ERA5 Data tool in UMEP is indeed in UTC. If by plotting you mean examining the data further, then I looked at the data earlier and the incoming shortwave radiation corresponds to the sunrise and sunset times at that location in UTC.

These are my data from Graz:
[Data_SEBE.zip](https://github.com/UMEP-dev/UMEP/files/7502967/Data_SEBE.zip)
As I wrote earlier, in my case the result makes much more sense to me if I don't fill in any UTC offset in SEBE. But I have no idea how exactly SEBE works with that data. Thanks for further examining.

**Reply by biglimp** [reply]
Sorry for being unclear regarding plotting. Yes, you can check with sunrise etc. to get a sense of UTC. If the metdata is in UTC=0, then you should set the UTC offset to 0 in SEBE. I plotted your data and it looks like it peaks at 11 am which indicates UTC=0 for your location. I ran your data with UTC=0 and I think it looks ok (see plots below). Another way to test the ERA5 data is to download data from e.g. Tokyo and see it that is also in UTC=0. I t will be easier to spot since Tokyo is further away. I started a download but I got queued.
![image](https://user-images.githubusercontent.com/11908356/140908128-c8ec82b9-b146-4ba3-a98a-f96bcf27cf1f.png)
![image](https://user-images.githubusercontent.com/11908356/140908233-c92157ed-ddd7-4fe0-a6d5-d8cde5995d09.png)

**Reply by MatLudvik** [reply]
Great, thanks a lot for your help! I got the same result. Good idea to try it on data far away. I tried to download other data as well to compare, but it failed while it was queued.
If it is as you say - that if you use ERA5 meteorological data obtained from the tool (Download data ERA5) in UMEP, which you get in UTC, then you have to leave the time offset at 0 (UTC) - would it be please possible to specify this in the UMEP Manual or Tutorial? I couldn't find this information there and that's why I was so confused. It only tells you to use time-offset if you are in a location with a different time zone, but it does not tell you that this is not the case while using the ERA5 data which not use the local time.

**Reply by biglimp** [reply]
Yes, sure. Please specify where you think this info is required in tutorials and manuals and I will put it in.

**Reply by MatLudvik** [reply]
In the [ERA5 chapter](https://umep-docs.readthedocs.io/en/latest/pre-processor/Meteorological%20Data%20Download%20data%20(ERA5).html), I would just mention that the data you will get from this tool will be in UTC.
And in the [SEBE chapter](https://umep-docs.readthedocs.io/en/latest/processor/Solar%20Radiation%20Solar%20Energy%20on%20Building%20Envelopes%20(SEBE).html), in the _UTC Offset (Hours)_ section, I would mention that if ERA5 data in UTC is retrieved for any area, the offset must match the downloaded meteo-data, i.e. remain in UTC as well. Of course, this would not apply to differently acquired data that are in local time, where the offset must then be set correctly. In the same chapter, in the _Input Meteorological File_ section is written, that the time should be in LST. If we calculated it correctly for a region of Graz in UTC+1 and we left the data in UTC and the offset also in UTC, in my opinion it would be a good idea to change this statement as well.
It might be worth mentioning this in the [SEBE tutorial](https://umep-docs.readthedocs.io/projects/tutorial/en/latest/Tutorials/SEBE.html) in the section _Preparing data for SEBE - Required inputs_ as well.
These are my suggestions, I hope it makes sense:)
